### DISCUSSION TITLE
Behavior for `from_items` input should be completely defined

### DISCUSSION THREAD

**Discussion by gibson042** [jmespath-community/jmespath.spec#123] [General]
See [discussion](https://github.com/jmespath-community/jmespath.spec/pull/102#discussion_r1008913623) in #102.

Behavior is currently undefined for `from_items` input that is not an array of [string, value] pairs. Examples:
1. items with length 0: ```from_items(`[ [] ]`)```
2. items with length 1: ```from_items(`[ ["keyWithNoValue"] ]`)```
3. items with length greater than 2: ```from_items(`[ ["key", "value", "extra"] ]`)```
4. items indicating non-string keys: ```from_items(`[ [1, "number"], [null, "null"], [{}, "object"] ]`)```

**Comment by springcomp** [comment]
I [would argue](https://github.com/jdevillard/JmesPath.Net/blob/master/src/jmespath.net/Functions/FromItemsFunction.cs#L20-L27):

- Input is expected to be an array of arrays, each of which has at least two values, the first being a string. If that is not the case, an error MUST be raised. The JMESPath.Net raises a new error type `invalid-value` but `invalid-type` would work equally well.
- Input can be an empty array, in which case, the result of `from_items()` is `{}` an empty object.

Therefore, examples 1, 2 and 4 raise an error.
Example 3 discards extra entries from each array and returns ` { "key": "value" } `

A case can be made for example 3 to also raise an error.

**Comment by gibson042** [comment]
I agree on rejecting 1, 2, and 4, and have a minor preference for rejecting 3 as well (aligning with the general strict input validation throughout the language, such as [sort](https://jmespath.site/main/#func-sort) rejecting mixed arrays).

**Reply by springcomp** [reply]
The more I think about this, the more I like the new `invalid-value` error type, here.

This could better apply to the `$count` parameter for `split()` and `replace()` as well for instance. When specifying a number, the syntax is correct and the type matches the signature. However, the function expects an integer.

What do you think?

**Reply by gibson042** [reply]
I'm in favor. 👍

**Reply by springcomp** [reply]
https://github.com/jmespath-community/jmespath.test/pull/13

**Reply by springcomp** [reply]
https://github.com/jmespath-community/jmespath.spec/pull/131
